OptimizelySDK/Odp/GraphQLManager.cs

```diff
@@ -94,15 +135,101 @@ public GraphQLManager(IErrorHandler errorHandler = null, ILogger logger = null,
         }
 
         /// <summary>
-        /// Parses JSON response
+        /// Converts the current QuerySegmentsParameters into a GraphQL query string
         /// </summary>
-        /// <param name="jsonResponse">JSON response from ODP</param>
-        /// <returns>Strongly-typed ODP Response object</returns>
-        public static Response ParseSegmentsResponseJson(string jsonResponse)
+        /// <returns>GraphQL payload</returns>
+        private static string ToGraphQLJson(string userKey, string userValue,
+            IEnumerable segmentsToCheck
+        )
+        {
+            var userValueWithEscapedQuotes = $"\\\"{userValue}\\\"";
+            var segmentsArrayJson =
+                JsonConvert.SerializeObject(segmentsToCheck).Replace("\"", "\\\"");
+
+            var json = new StringBuilder();
```

Comment:
can we revise this approach?
e.g. 
Function should take children. and append the children inside function. "Closing and Opening brackets" may cause issues in this kind of situation.